### PR TITLE
Fix 'override' and 'override default' functionality

### PR DESCRIPTION
**Pull request by iilyak**
COUCHDB-2788

### CODE REVIEW

File: src/chttpd_handlers.erl [hunk 1]
```diff
@@ -55,11 +55,39 @@ select([], Default) ->
...
+    do_select(Rest, Acc);
+do_select([Handler], Acc) ->
+    [Handler | Acc];
+do_select([Handler | Rest], Acc) ->
+    do_select(Rest, [Handler | Acc]).
+
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+
+select_override_test_() ->
```
Comment (rnewson) [inline src/chttpd_handlers.erl]: why are these test generators rather than direct tests?

Comment (iilyak) [inline src/chttpd_handlers.erl]: @rnewson: Fixed

**Comment by rnewson** [comment]
+1 after squash

**Comment by iilyak** [comment]
PR is squashed.

**Comment by rnewson** [comment]
@iilyak then merge? :)

**Comment by iilyak** [comment]
Yes it is good to merge.

**Pull request merged**